### PR TITLE
updates to try to make appimage work

### LINKED ISSUE #42: Provide official Linux artifacts
We need to decide if we want to build this statically or dynamically. We SHOULD be able to provide a static build with Qt, so long as it can be recompiled against another version, but unclear what makes the most sense. It's probably safe to build this on ubuntu but we should probably do some research into how people distribute official linux binaries. As always, users are free to compile from source.

### PR DESCRIPTION
**Pull request by JoelAtDeluxe**
This PR adds the ability for app image generation when a pull request is made.

Note: this uses the linuxdeployqt binary to create an appimage. This tool requires you use the oldest LTS version of ubuntu. Currently that is  ubuntu 16, however, around april, that should shift to Ubuntu 18, which should be good for another 2 years after that. 

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by JoelAtDeluxe** [comment]
I believe that once the code in PR #84 is merged that this should begin building correctly

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -39,6 +40,43 @@ jobs:
       with:
         submodules: true
 
+    - name: Cache Qt
```
Comment (jkennedyvz) [inline .github/workflows/ci.yaml:43]: Did we determine if this is necessary or not?

**Comment by jkennedyvz** [comment]
We should add a linux matrix item [here](https://github.com/theparanoids/ashirt/blob/js-linux-appimage/.github/workflows/ci.yaml#L167) for release

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (JoelAtDeluxe) [reply .github/workflows/ci.yaml:43]: it shouldn't be necessary, but it should speed up CI. I haven't timed it though

**Review by jkennedyvz — APPROVED** [review]
Looks great Joel!

**Pull request merged**